src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -380,18 +380,30 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 	{
 #if PG_VERSION_NUM >= PG_VERSION_16
 		DefElem *opt = NULL;
+		 bool appended_option = false; // To track if any option has been appended
 		foreach_ptr(opt, stmt->opt)
 		{
 			if (strcmp(opt->defname, "admin") == 0)
 			{
 				appendStringInfo(buf, " WITH ADMIN OPTION");
-				break;
+				appended_option = true;
+				// break; // exits after finding "admin" (is this needed?) 
+			}
+			else if (strcmp(opt->defname,"inherit")==0)
+			{
+				if (appended_option) appendStringInfo(buf, " ");
+            appendStringInfo(buf, "WITH INHERIT TRUE");
+            appended_option = true;
 			}
+			else if((strcmp(opt->defname, "set") == 0)){
```

Comment:
Here you could also have the following: `GRANT x TO y WITH INHERIT TRUE, SET TRUE;`
The `WITH` clause is only added once for the first option, and then the others follow with comma. So, it would be good to change the logic in this loop a bit.

Also, please keep in mind the default values for these options: https://www.postgresql.org/docs/devel/sql-grant.html

> The ADMIN option allows the member to in turn grant membership in the role to others, and revoke membership in the role as well. Without the admin option, ordinary users cannot do that. A role is not considered to hold WITH ADMIN OPTION on itself. Database superusers can grant or revoke membership in any role to anyone. This option defaults to FALSE.

> The INHERIT option, if it is set to TRUE, causes the member to inherit the privileges of the granted role. That is, it can automatically use whatever database privileges have been granted to that role. If set to FALSE, the member does not inherit the privileges of the granted role. If this clause is not specified, it defaults to true if the member role is set to INHERIT and to false if the member role is set to NOINHERIT. See [CREATE ROLE](https://www.postgresql.org/docs/devel/sql-createrole.html).

> The SET option, if it is set to TRUE, allows the member to change to the granted role using the [SET ROLE](https://www.postgresql.org/docs/devel/sql-set-role.html) command. If a role is an indirect member of another role, it can use SET ROLE to change to that role only if there is a chain of grants each of which has SET TRUE. This option defaults to TRUE.